### ISSUE TITLE
Cast to SOAPDocument may fail in ElementImpl.getOwnerDocument()

### ISSUE THREAD

**Issue by Tomas-Kraus** [eclipse-ee4j/metro-saaj#54]
I am using Java 5, Apache CXF 2.1.10, WSS4J 1.5.8 and OpenSAML 1.1 to produce an
outgoing SOAP message with a signed SAML token in it.

This fails with a ClassCastException, and I traced it to the following method:
com.sun.xml.messaging.saaj.soap.impl.ElementImpl.getOwnerDocument().
Specifically, the following unchecked cast fails:

SOAPDocument ownerSOAPDocument =
((SOAPDocument) super.getOwnerDocument());

The reason it fails is because the ElementImpl object was previously processed
by org.w3c.dom.Document.adoptNode(Node) call, which changed the owner document
of the element, which is legal. This happens in
org.opensaml.SAMLSubject.toDOM(Document, boolean), line 341:

conf.appendChild(doc.adoptNode(keyInfo.getElement()));

Long story short, I'd guess that aforementioned cast needs to be checked; I am
currently unable to use SAAJ with OpenSAML to produce SAML tokens.
#### Environment
Operating System: All
Platform: All
#### Affected Versions
[1.4]
